code-studio/src/js/pd/workshop_dashboard/components/workshop_form.jsx

```diff
@@ -128,6 +132,24 @@ var WorkshopForm = React.createClass({
   },
 
   enableAutocompleteLocation: function () {
+    if (!this.state.useAutocomplete) {
+      return;
+    }
+
+    // The way to catch google auth errors is in a global function :(
+    // See https://developers.google.com/maps/documentation/javascript/events#auth-errors
+    // If google auth fails, remove the autocomplete and re-draw.
+    if (!this.gm_authFailure) {
+      // Save existing function, if one exists.
+      this.old_gm_authFailure = window.gm_authFailure;
+      window.gm_authFailure = this.gm_authFailure = () => {
+        if (this.old_gm_authFailure) {
+          this.old_gm_authFailure();
```

Comment:
Maybe we don't? Since it's a global, I wanted to make sure I wasn't blowing away an existing function. See line 98 above. On dismount I set it back to the old one.